clicache/integration-test2/GarbageCollectCache.cs

```diff
@@ -0,0 +1,177 @@
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Apache.Geode.Client.IntegrationTests
+{
+    [Trait("Category", "Integration")]
+    public class GarbageCollectCache : TestBase
+    {
+        public GarbageCollectCache(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+        }
+
+        [Fact]
+        public void VerifyNoLeakedThreads()
+        {
+            using (var cluster = new Cluster(output, CreateTestCaseDirectoryName(), 1, 1))
+            {
+                Assert.True(cluster.Start());
+                Assert.Equal(0, cluster.Gfsh
+                    .create()
+                    .region()
+                    .withName("testRegion")
+                    .withType("PARTITION")
+                    .execute());
+
+                for (int i=0; i<25; i++)
+                {
+                    var ncThreadsBefore = Process.GetCurrentProcess().Threads.Count;
+
+                    using (var cache = new CacheFactory()
+                        .Set("log-level", "none")
+                        .Create())
+                    {
+
+                        cluster.ApplyLocators(cache.GetPoolFactory()).Create("default");
+
+                        var regionFactory = cache.CreateRegionFactory(RegionShortcut.PROXY)
+                            .SetPoolName("default");
+
+                        var region = regionFactory.Create<string, string>("testRegion");
+
+                        const string key = "hello";
+                        const string expectedResult = "dave";
+
+                        region.Put(key, expectedResult);
+                        var actualResult = region.Get(key);
+                        Assert.Equal(expectedResult, actualResult);
+
+                        cache.Close();
+                    }
+
+                    var ncThreadsAfter = Process.GetCurrentProcess().Threads.Count;
+                    var ratio = ncThreadsBefore / (double)ncThreadsAfter;
+
+                    // Because the number of threads in the process depends on
+                    // the number of cores, debug vs release, whether the GC thread
+                    // is running, etc., a robust test is to check whether the ratio
+                    // of before and after threads is close to one. Also skipping the
+                    // first couple of iterations avoids threads related to test 
+                    // environment startup.
+                    if (i > 5)
+                    {
+                        //Assert.True(.8 < ratio && ratio < 1.3);
+                        string error = "ncThreadsBefore = " + ncThreadsBefore.ToString() +
+                            ", ncThreadsAfter = " + ncThreadsAfter.ToString();
+                        Assert.False(!(.8 < ratio && ratio < 1.3), error);
+                    }
+                }
+            }
+        }
+
+        [Fact(Skip = "Need a heuristic to filter out GC effect.")]
```

Comment:
Added a skipped test? What does this do for us?

Reply:
Removing until we can come up with a heuristic that works properly in the CI.